examples/bsoncxx/builder_stream.cpp

```diff
@@ -73,18 +73,24 @@ int main(int, char**) {
 
     // To nest an existing bsoncxx::document::value into a builder stream, you can create a
     // types::b_document and append it. Alternatively you can open a new document and concatenate
-    // the value in.
+    // the value in. Or, simplest, is just to stream in the view.
     bsoncxx::document::value nestedValue = document{} << "nested" << true << finalize;
     document topLevelDoc{};
-    topLevelDoc << "subDoc1" << types::b_document{nestedValue.view()} << "subDoc2" << open_document
-                << concatenate(nestedValue.view()) << close_document;
+    auto state1 = topLevelDoc << "subDoc1" << types::b_document{nestedValue.view()};
+    auto state2 = state1 << "subDoc2" << open_document << concatenate(nestedValue.view()) << close_document;
+    auto state3 = state2 << "subDoc3" << nestedValue.view();
+    auto final = state2 << finalize;
```

Comment:
I don't think breaking this up into multiple states is the clearest way to convey this.   Having multiple states means I'm wondering why there are multiple states instead of focusing on three techniques.  While not "proper" formatting, I think something like this would be better:
```
topLevelDoc
    << "subDoc1" << types::b_document{nestedValue.view()}
    << "subDoc2" << open_document << concatenate(nestedValue.view())
    << "subDoc3" << nestedValue.view()
    << close_document;
```

This is sort of a "Tufte" take on the example (i.e. 'reduce non-data ink').